nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessGroupResource.java

```diff
@@ -4597,6 +4611,246 @@ public Response deleteReplaceProcessGroupRequest(
         return deleteFlowUpdateRequest("replace-requests", replaceRequestId, disconnectedNodeAcknowledged.booleanValue());
     }
 
+    // -------------
+    // flow-analysis
+    // -------------
+
+    /**
+     * Submits a request to run a flow analysis.
+     *
+     * @param processGroupId The id of the process group representing (a part of) the flow to be analyzed
+     * @return An AnalyzeFlowRequestEntity
+     */
+    @POST
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("flow-analysis/{processGroupId}")
```

Comment:
In the context of this Process Group Resource, the path should include the ID first, followed by `flow-analysis-requests`:
```suggestion
    @Path("{id}/flow-analysis-requests")
```